src/Http/Controllers/Baskets/BasketLineController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace GetCandy\Api\Http\Controllers\Baskets;
+
+use Illuminate\Support\Facades\Request;
+use GetCandy\Api\Http\Controllers\BaseController;
+use GetCandy\Api\Http\Resources\Baskets\BasketResource;
+use GetCandy\Api\Core\Baskets\Services\BasketLineService;
+use GetCandy\Api\Http\Requests\Baskets\UpdateLineRequest;
+use GetCandy\Api\Core\Baskets\Factories\BasketLineFactory;
+use GetCandy\Api\Http\Requests\Baskets\CreateLinesRequest;
+use GetCandy\Api\Http\Requests\Baskets\DeleteLinesRequest;
+use GetCandy\Api\Http\Requests\Baskets\ChangeQuantityRequest;
+
+
+class BasketLineController extends BaseController
+{
+    /**
+     * @var BasketLineFactory
+     */
+    protected $factory;
+
+    /**
+     * @var BasketLineService $basketLines
+     */
+    protected $basketLines;
+
+    public function __construct(BasketLineFactory $factory)
+    {
+        $this->factory = $factory;
+        $this->basketLines = app('api')->basketLines();
+        $this->basketLines->setIncludes(Request::get('includes') ?? '');
+    }
+
+    /**
+     * Store one or more new basket lines, and associate them with a basket ID.
+     *
+     * @param CreateLinesRequest $request
+     *
+     * @return array|BasketResource
+     */
+    public function store(CreateLinesRequest $request)
+    {
+        try {
+            $basket = app('api')->baskets()->addLines($request->all(), $request->user());
+        } catch (\Illuminate\Database\QueryException $e) {
+            return $this->errorUnprocessable(trans('getcandy::validation.max_qty'));
+        }
+
+        return new BasketResource($basket);
+    }
+
+    /**
+     * Update a basket line's quantity.
+     *
+     * @param string $id
+     * @param UpdateLineRequest $request
+     *
+     * @return BasketResource
+     */
+    public function update(string $id, UpdateLineRequest $request)
+    {
+        $quantity = $request['quantity'];
```

Comment:
Can we move from referencing arrays to objects so `$request->quantity` we could also shorten this and pass the reference straight into the `setQuantity` method.

Reply:
Sure thing!